src/app/api/check-stale-token/checkStaleToken.controller.ts

```diff
@@ -0,0 +1,25 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '@api/core/utils/authenticate'
+import { z } from 'zod'
+import APIError from '@api/core/exceptions/api'
+import httpStatus from 'http-status'
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import { CheckStaleTokenResponse } from '@/types/dto/checkStaleToken.dto'
+
+export const checkStaleToken = async (req: NextRequest) => {
+  const user = await authenticate(req)
+  const { data: clientId, error: clientIdError } = z.string().safeParse(user.clientId)
```

Comment:
string destructured as object? is this correct?

Reply:
I'm using `safeParse` here, which returns an object with keys data, success and error.